File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    
    while(true){
        if(priorities[0] == Math.max(...priorities)){
            answer++;
            if(location == 0){
                break;
            }
            priorities.shift();
            location--;
        }
        else{
            let num = priorities.shift();
            priorities.push(num);
            if(location == 0){
                location = priorities.length - 1;
            }
            else{
                location--;
            }
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        System.out.println(Arrays.stream(priorities).max().getAsInt());
       
        int idx = 0;
        while(true){
            if(Arrays.stream(priorities).max().getAsInt() == priorities[idx]){ 
                answer++;
                if(idx == location) break;
                else{
                    priorities[idx] = 0;
                }
            }
            
            idx++;
            if(idx == priorities.length) idx=0;
        }
        
        return answer;
    }
}